phper/src/builder/data_type.rs

```diff
@@ -0,0 +1,43 @@
+pub enum DataType {
+    // PHP < 8.2: mixed
+    // PHP >= 8.2: null
+    Null,
+    // T1|T2|T3...
+    Union(Vec<DataType>),
+    // float
+    Float,
+    // int
+    Integer,
+    // int
+    PostivieInteger,
+    // int
+    RangedInteger(i64, i64),
+    // string
+    String,
+    // string
+    NonEmptyString,
+    // string
+    Classname,
+    // bool
+    Boolean,
+    // true
+    True,
+    // false
+    False,
+    // array
+    HashTable,
+    // array
+    List(Vec<DataType>),
+    // array
+    Array(Vec<DataType>),
```

Comment:
types such as `PostivieInteger`, `RangedInteger`, `NonEmptyString`, `List`, `Array` and `Classname` are not "real" types, they get simplified by phper to the their PHP equivalent, however, we should use this when parsing the argument, this will simplify things for extension authors, this information should also be used when generating php stubs.

e.g: 

when generating stub for a function with argument `a` of type `Classname`, we should refer to it as `string` in the function signature, but we should add `@psalm-param class-string $a` and `@phpstan-param class-string $a` to the phpdoc, so static analysis tools can warn users that pass a non-class string to to `$a`.

Reply:
atomic types from psalm: https://psalm.dev/docs/annotating_code/type_syntax/atomic_types/